lesson_09/test_homework_09.py

```diff
@@ -0,0 +1,100 @@
+import unittest
+from homework_09 import (
+    multiplication_table,
+    fibonacci_generator_solution,
+    is_palindrome,
+    find_max_length_palindrome,
+)
+
+
+class TestSomeFunctions(unittest.TestCase):
+
+    def test_multiplication_table_positive(self):
+        """ Multiplication positive test. Positive integer number as function argument. """
+        actual_result = multiplication_table(5)
+        expected_result = [5, 10, 15, 20, 25]
+        self.assertEqual(actual_result.pop(), expected_result.pop())
+        self.assertEqual(len(actual_result), len(actual_result))
+
+    def test_multiplication_table_negative(self):
+        """ Multiplication negative test. Negative integer number as function argument. """
+        test_number = -4
+        actual_result = multiplication_table(test_number)
+        expected_result = f"'{test_number}' passed to function, must be positive"
+        self.assertEqual(actual_result, expected_result)
+
+    def test_multiplication_table_incorrect_datatype(self):
+        actual_result = multiplication_table([5])
+        expected_result = f"Instance must be 'int' or 'float' type, current instance: 'list'"
+        self.assertEqual(actual_result, expected_result)
```

Comment:
```suggestion
        expected_result = f"Instance must be 'int' or 'float' type, current instance: 'list'"
        self.assertEqual(actual_result, expected_result)
```
ще один явний  кандидат у ексепшин

Reply:
Звичайно, я переробив функцію - і тепер під неї переробив тест с ексепшином